usb1/__pyinstaller/hook-usb1.py

```diff
@@ -0,0 +1,8 @@
+import logging
+from PyInstaller.utils.hooks import collect_dynamic_libs
+
+logger = logging.getLogger(__name__)
+logger.info("--- libusb1 pyinstaller hook ---")
+binaries = collect_dynamic_libs('usb1')

```

Comment:
To confirm my understanding: Is this needed mostly because of the archaic way I locate the embedded dll, which requires the egg to be on the filesystem ? I tried `importlib.resource` but sadly its backport does not seem to work on <3.3 . When I drop 2.7 support, I think I will switch to that, and then this may lead to further simplification in these hooks - if this is really what they are doing.

Reply:
pyinstaller just analyzes the imports of the project which is about to be bundled. As the dynamic librarys are not imported directly they will be missing. This line just tells pyinstaller to look for all dynamic libraries in the usb1 package.

Reply:
Oh, yes, this part I understood. I rather wonder what pyinstaller does with that dll once when the bundled application is being run: my code will either try to load the dll from the `usb1` module's path, or by asking the OS' linker to locate it. I expect the former to only work if the module and the dll are extracted in the file system (and not, say, imported from a resource section of the executable with magic import hooks).

Reply:
PyInstaller can work in two modes: Single Directory Mode or Single File Mode. The later is just a self extracting archive which is uncompressed to a temporary location and executed from there. So as long as pyinstaller is informed that a given dll must be included in the build it should work as the library is placed at the right location in the `dist` folder.

Reply:
Makes sense, thanks. So it will reliably work like this.